// if key == value, it tells `getErrorMessage` to keep the original message
const errors = {
    1001: 'Internal server error',
    1021: 'Nickname already in use',
    1025: 'The given token is invalid',
    1027: 'Invalid credentials',
    1030: 1030,
    1031: 1031,
    1032: 1032,
    1033: 1033,
    1041: 1041,
}

export const getErrorMessage = (error) => {
    // error not generated by api
    if (!(error instanceof Object) || !('code' in error)) {
        console.error(error)
        return 'Unexpected error'
    }

    // if code is not known
    if (!(error.code in errors)) {
        console.error(error)
        return 'Unknown error'
    }

    if (error.code == errors[error.code]) {
        return error.message
    }

    return errors[error.code]
}
